app/src/main/java/com/github/se/eventradar/ui/viewProfile/ProfileUi.kt

```diff
@@ -113,68 +161,183 @@ fun ProfileUi(
+
+                Button(
+                    onClick = {
+                      if (viewModel.validateFields()) {
+                        viewModel.updateUserInfo()
+                        isEditMode.value = false
+                      }
+                    },
+                    modifier = Modifier.padding(top = 10.dp).testTag("saveButton")) {
+                      Text("Save")
+                    }
+              } else {
+                NameText(
+                    name = "${uiState.firstName} ${uiState.lastName}",
+                    modifier = Modifier.testTag("name"))
+                StandardProfileInformationText(
+                    text = "@${uiState.username}", modifier = Modifier.testTag("username"))
+                Column(
+                    modifier = Modifier.fillMaxWidth().testTag("leftAlignedViewProfileColumn"),
+                ) {
+                  Text(
+                      text = "Bio",
+                      modifier = Modifier.padding(top = 16.dp).testTag("bioLabelText"),
+                      color = MaterialTheme.colorScheme.onSurfaceVariant,
+                      fontSize = 12.sp,
+                      style = MaterialTheme.typography.labelMedium)
+                  Text(
+                      text = uiState.bio,
+                      modifier = Modifier.padding(top = 4.dp).testTag("bioInfoText"),
+                      color = MaterialTheme.colorScheme.onSurface,
+                      fontSize = 16.sp,
+                      style = MaterialTheme.typography.bodyMedium)
+                  if (!isPublicView) {
+                    Row(
+                        modifier = Modifier.padding(top = 16.dp).testTag("phoneNumberBirthDateRow"),
+                        horizontalArrangement = Arrangement.SpaceBetween) {
+                          Column(modifier = Modifier.weight(1f).testTag("phoneNumberColumn")) {
+                            Text(
+                                text = "Phone Number",
+                                modifier =
+                                    Modifier.padding(top = 16.dp).testTag("phoneNumberLabelText"),
+                                color = MaterialTheme.colorScheme.onSurfaceVariant,
+                                fontSize = 12.sp,
+                                style = MaterialTheme.typography.labelMedium)
+                            Text(
+                                text = uiState.phoneNumber,
+                                modifier =
+                                    Modifier.padding(top = 4.dp).testTag("phoneNumberInfoText"),
+                                color = MaterialTheme.colorScheme.onSurface,
+                                fontSize = 16.sp,
+                                style = MaterialTheme.typography.bodyMedium)
+                          }
+                          Column(modifier = Modifier.weight(1f).testTag("birthDateColumn")) {
+                            Text(
+                                text = "Birth Date",
```

Comment:
suggestion: could add a placeholder for the birth date format here :)

Reply:
Perhaps not a placeholder, since the existing DOB text is displayed, but add an example below the text input, same as what is done for username having the text "Your username must be unique". Text can be "e.g. 10.12.1997"

Reply:
Good idea! Agree, that would be more user-friendly